keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -219,15 +258,42 @@ open class SqlResourceRepository(
     }
   }
 
+  override fun appendHistory(event: ApplicationEvent) {
+    if (event.ignoreRepeatedInHistory) {
+      val previousEvent = jooq
+        .select(EVENT.JSON)
+        .from(EVENT)
+        .where(EVENT.SCOPE.eq(Scope.APPLICATION.name))
+        .and(EVENT.UID.eq(event.application))
+        .orderBy(EVENT.TIMESTAMP.desc())
+        .limit(1)
+        .fetchOne()
+        ?.let { (json) ->
+          objectMapper.readValue<ApplicationEvent>(json)
+        }
+
+      if (event.javaClass == previousEvent?.javaClass) return
+    }
+
+    jooq
```

Comment:
General question about code with multiple database queries: do we wrap them in transactions?

Reply:
We've been doing that more and more and filling in the gaps. I can add that.